File: JULY-2025/29-07-2025/p2411.scala
object Solution {
  import scala.util.chaining._
  import collection.immutable.BitSet
  def smallestSubarrays(nums: Array[Int]): Array[Int] =
    nums.toList.zipWithIndex.reverseIterator.scanLeft((Map.empty[Int,Int],0)){case ((aMap,acc),(num,i))=>
      (aMap ++ BitSet.fromBitMask(Array(num)).map(_ -> i), acc | num)
    }.drop(1).toList.reverse.pipe(sfx => (nums.indices.toArray zip sfx).map{case (i,(bMap,mx)) =>
      if(mx == 0) 1 else BitSet.fromBitMask(Array(mx)).iterator.map(bMap).max - i + 1
    })
}

File: JULY-2025/29-07-2025/p2411.go
func smallestSubarrays(nums []int) []int {
    var bitmap [30]int

    for i := len(nums) - 1; i >= 0; i-- {
        for j := 0; (1 << j) <= nums[i]; j++ {
            if nums[i] & (1 << j) != 0 {
                bitmap[j] = i
            }
        }
        nums[i] = max(bitmap[:], i) - i + 1
    }

    return nums
}

func max(arr []int, i int) int {
    max := i
    for _, x := range arr {
        if x > max {
            max = x
        }
    }
    return max
}